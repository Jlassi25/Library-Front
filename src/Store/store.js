import {  configureStore } from "@reduxjs/toolkit";
import {  CategoriesSlice } from "../features/category/categorySlice";




export const store = configureStore({
    reducer: {
        CategoriesSlice:CategoriesSlice.reducer
    }
})