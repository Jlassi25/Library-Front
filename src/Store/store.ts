import {  configureStore } from "@reduxjs/toolkit";
import {  CategoriesSlice } from "../features/category/categorySlice";
import {  subscriberSlice } from "../features/subscriber/subscriberSlice";




export const store = configureStore({
    reducer: {
        CategoriesSlice:CategoriesSlice.reducer,
        Subscribers:subscriberSlice.reducer
    }
})

export type AppDispatch = typeof store.dispatch