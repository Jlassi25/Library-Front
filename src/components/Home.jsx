

const Home = () => {
  return (
    < >
      <div className="container-fluid">
        <div className="section">
          <div className="row">
            <div className="col-6 col-md-2" />
            <div className="col-6 col-md-8">
              <br /><br />
              <h1 style={{ textAlign: 'center', fontWeight: 400, color: 'rgb(2, 23, 5)' }}>
                Hello and welcome back Admin
              </h1>
              <br /><br />
              <h2 style={{ textAlign: 'center', fontWeight: 300, color: 'rgb(2, 23, 5)' }}>
                Here's some statistics of Vote.ly
              </h2>
            </div>
            <div className="col-6 col-md-2" />
          </div>
          <br />
          <div className="row">
            <div className="col-6 col-md-3" />
            <div className="col-6 col-md-3">
              <div className="card card-stats">
                <div className="card-header card-header-success card-header-icon" style={{ backgroundColor: '#d8f3dc' }}>
                  <div className="card-icon">
                    <i className="nc-icon nc-icon nc-chart-bar-32" />
                  </div>
                  <p className="card-category" style={{ color: 'black' }}>
                    Number of created competitions
                  </p>
                  <a className="nav-link" href="home">
                    <h4 className="card-title"><b>43</b> competitions</h4>
                  </a>
                </div>
              </div>
            </div>
            <div className="col-6 col-md-3">
              <div className="card card-stats">
                <div className="card-header card-header-success card-header-icon" style={{ backgroundColor: '#d8f3dc' }}>
                  <div className="card-icon">
                    <i className="nc-icon nc-icon nc-chart-pie-35" />
                  </div>
                  <p className="card-category" style={{ color: 'black' }}>
                    Number of registred users
                  </p>
                  <a className="nav-link" href="home">
                    <h4 className="card-title"><b>20</b> users</h4>
                  </a>
                </div>
              </div>
            </div>
            <div className="col-6 col-md-3" />
          </div>
        </div>
      </div>
      {/* what you looking to do*/}
      <br /><br />
      <div className="container-fluid">
        <div className="section">
          <div className="row">
            <div className="col-6 col-md-3" style={{ textAlign: 'center' }} />
            <div className="col-6 col-md-6" style={{ textAlign: 'center' }}>
              <h2 style={{ textAlign: 'center', fontWeight: 300, color: 'rgb(2, 23, 5)' }}>
                What are you looking to do today?
              </h2>
            </div>
            <div className="col-6 col-md-3" style={{ textAlign: 'center' }} />
          </div>
        </div>
      </div>
      <br />
      <div className="container-fluid">
        <div className="section">
          <div className="row">
            <div className="col-6 col-md-4" style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
              <div className="box-1">
                <div className="btnn btn-one">
                  <span>Manage competitions</span>
                </div>
              </div>
            </div>
            <div className="col-6 col-md-4" style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
              <div className="box-1">
                <div className="btnn btn-one">
                  <span>Validate competition</span>
                </div>
              </div>
            </div>
            <div className="col-6 col-md-4" style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
              <div className="box-1">
                <div className="btnn btn-one">
                  <span>Manage users</span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      {/* li ta7ti tsakkar el content sakarha ekhir el page*/}
    </>





  );
}

export default Home;