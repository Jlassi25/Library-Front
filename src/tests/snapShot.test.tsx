import { describe, it, expect, vi } from "vitest";
import { render, screen, waitFor } from '@testing-library/react';
import { Provider } from "react-redux";
import { store } from "../Store/store";

import React from "react";
import Home from "../components/Home";
import { MemoryRouter } from "react-router-dom";

//SnapShot  TEST
describe('Home Page', () => {
    it('renders snapshot', () => {
        const { container } = render(  
        <Provider store={store}>
                 <MemoryRouter>
                 <Home/>
                 </MemoryRouter>
           
        </Provider>);
        expect(container.firstChild).toMatchSnapshot();
    });
});