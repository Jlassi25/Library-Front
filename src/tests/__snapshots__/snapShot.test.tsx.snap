// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page > renders snapshot 1`] = `
<div
  class="container-fluid"
>
  <div
    class="section"
  >
    <div
      class="row"
    >
      <div
        class="col-6 col-md-2"
      />
      <div
        class="col-6 col-md-8"
      >
        <br />
        <br />
        <h1
          style="text-align: center; font-weight: 400; color: rgb(2, 23, 5);"
        >
          Hello and welcome back Admin
        </h1>
        <br />
        <br />
        <h2
          style="text-align: center; font-weight: 300; color: rgb(2, 23, 5);"
        >
          Here's some statistics of Library
        </h2>
      </div>
      <div
        class="col-6 col-md-2"
      />
    </div>
    <br />
    <div
      class="row"
    >
      <div
        class="col-6 col-md-3"
      />
      <div
        class="col-6 col-md-3"
      >
        <div
          class="card card-stats"
        >
          <div
            class="card-header card-header-success card-header-icon"
            style="background-color: rgb(216, 243, 220);"
          >
            <div
              class="card-icon"
            >
              <i
                class="nc-icon nc-icon nc-chart-bar-32"
              />
            </div>
            <p
              class="card-category"
              style="color: black;"
            >
              Number of Book
            </p>
            <a
              class="nav-link"
              href="/book"
            >
              <h4
                class="card-title text-center"
              >
                <b />
                 Books
              </h4>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-6 col-md-3"
      >
        <div
          class="card card-stats"
        >
          <div
            class="card-header card-header-success card-header-icon"
            style="background-color: rgb(216, 243, 220);"
          >
            <div
              class="card-icon"
            >
              <i
                class="nc-icon nc-icon nc-chart-pie-35"
              />
            </div>
            <p
              class="card-category"
              style="color: black;"
            >
              Number of Subscriber
            </p>
            <a
              class="nav-link"
              href="/subscriber"
            >
              <h4
                class="card-title text-center"
              >
                <b />
                 Subscribers
              </h4>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-6 col-md-3 mt-5 d-flex justify-content-center align-items-center"
      />
      <div
        class="col-6 col-md-12 d-flex justify-content-center align-items-center"
      >
        <div
          class="card card-stats"
        >
          <div
            class="card-header card-header-success card-header-icon"
            style="background-color: rgb(216, 243, 220);"
          >
            <div
              class="card-icon"
            >
              <i
                class="nc-icon nc-icon nc-chart-pie-35"
              />
            </div>
            <p
              class="card-category"
              style="color: black;"
            >
              Number of bookings
            </p>
            <a
              class="nav-link"
              href="/subscriber"
            >
              <h4
                class="card-title text-center"
              >
                <b />
                 
              </h4>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
